lib/py-ios-device-client.js

```diff
@@ -0,0 +1,80 @@
+import { exec } from 'teen_process';
+import { fs, util, tempDir } from 'appium-support';
+import log from './logger';
+import path from 'path';
+
+// https://github.com/YueChen-C/py-ios-device
+
+const BINARY_NAME = 'pyidevice';
+
+class Pyidevice {
+  constructor (udid) {
+    this.udid = udid;
+    this.binaryPath = null;
+  }
+
+  async assertExists (isStrict = true) {
+    if (this.binaryPath) {
+      return true;
+    }
+
+    try {
+      this.binaryPath = await fs.which(BINARY_NAME);
+      return true;
+    } catch (e) {
+      if (isStrict) {
+        throw new Error(`${BINARY_NAME} binary cannot be found in PATH. ` +
+          `Please make sure it is installed. Visit https://github.com/YueChen-C/py-ios-device for ` +
+          `more details.`);
+      }
+      return false;
+    }
+  }
+
+  async execute (args, ensureFormat = true) {
+    await this.assertExists();
+
+    const finalArgs = ['--udid', this.udid, ...args];
```

Comment:
`--udid` should be after commands such as `profiles`. So, `[...args, '--udid', this.udid];`

Reply:
fixed